### PR TITLE
add `check_environments_match` function for checking if two environments are identical

### PR DESCRIPTION
**Pull request by Kallinteris-Andreas**
# Description

implements https://github.com/Farama-Foundation/Gymnasium/issues/575

### CODE REVIEW

File: gymnasium/utils/env_tests.py
Comment (pseudo-rnd-thoughts) [inline gymnasium/utils/env_tests.py:1]: Rename to `env_match`

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Could you update the mujoco version test to use this function?

**Comment by Kallinteris-Andreas** [comment]
`test_mujoco_v3` does not check for equality it checks for close "enought", which this function does not support

File: gymnasium/utils/env_tests.py
Comment (pseudo-rnd-thoughts) [reply gymnasium/utils/env_tests.py:1]: Sorry, I meant the filename rather the function name, I think `check_environments_match` is a good name

**Review by pseudo-rnd-thoughts — APPROVED** [review]
My only worry is that we don't have any testing but I don't imagine this being a popular function

**Pull request merged**